File: Array/Rotate Image/Rotate image.java
class Solution {
    public void rotate(int[][] matrix) {
        int colms = matrix[0].length;
        int i = 0, j = 0;
        int [][] rotate = new int [colms][colms];
        for(int col = 0; col < colms; col++){
            for(int r = matrix.length - 1; r >= 0; r--){
                rotate[i][j] = matrix[r][col];
                j++;
            }
            i++;
            j = 0;
        }
         for(int col = 0; col < colms; col++){
            for(int r = 0; r < colms; r++){
                matrix[col][r] = rotate[col][r];
            }
         }
        
    }
}

File: Array/Rotate Image/Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        vector<int> v;
        vector< vector<int>> rotate;
        int colms = matrix[0].size();
        for(int col = 0; col < colms; col++){
            for(int r = matrix.size() - 1; r >= 0; r--){
                v.push_back(matrix[r][col]);
            }
            rotate.push_back(v);
            v.clear();
        }
        matrix = rotate;
    }
};
